source/images.py

```diff
@@ -0,0 +1,129 @@
+from typing import List
+
+import numpy as np
+from PIL import Image
+
+
+class Images:
+    """Module to handle image conversion and processing"""
+
+    def __init__(self, file: str,
+                 cols: int = 80,
+                 scale: float = 0.43,
+                 resolution: int = 0,
+                 shade_str: str = "$@B%8&WM#*oahkbdpqwmZO0QLCJUYXzcvunxrjft/\\()1{}[]?-_+~<>i!lI;:,\"^`'.",
+                 shade_min: str = "@#+-."):
+        """
+        Initiates the Images class while takes an image file and converts it into an Ascii image
+
+        :param file: The image file path, allows .jpg, .jpeg, .png and other popular file formats
+        Ascii related parameters =>
+        :param cols: Number of columns in the ascii image, defaults to 80
+        :param scale: defines the adjusted aspect ratio for the ascii image, defaults to 0.43
+        :param resolution: defines resolution, takes in values as percentage of the highest resolution possible with
+               characters from shade_str
+               Special value: 0 (default) is interpreted as mode switch to minimalistic output with characters from
+               shade_min
+        :param shade_str: defaults to a well known pattern of 69 characters used to define the greyscale
+        :param shade_min: defaults to a minimalistic pattern of 5 characters used to define the greyscale
+        """
+        # TODO: error handling when no file is selected
+        self.file = file
+        self.image = Image.open(file).convert("L")
+        self.cols = cols
+        self.scale = scale
+        self.shade = {
```

Comment:
Why do you need a dict of this btw?

Reply:
I wanted to keep all shade related variables together, it got a little messy to manage that many separate variables

Reply:
Good point